central/graphql/resolvers/nodes.go

```diff
@@ -313,28 +330,66 @@ func (resolver *nodeResolver) ComponentCount(ctx context.Context, args RawQuery)
 	return resolver.root.componentCountV2(resolver.nodeScopeContext(ctx), RawQuery{Query: &query})
 }
 
+// NodeComponents returns the components in the node.
+func (resolver *nodeResolver) NodeComponents(ctx context.Context, args PaginatedQuery) ([]NodeComponentResolver, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.Nodes, "NodeComponents")
+	if err := readNodes(ctx); err != nil {
+		return nil, err
+	}
+
+	if !features.PostgresDatastore.Enabled() {
+		query := search.AddRawQueriesAsConjunction(args.String(), resolver.getNodeRawQuery())
+		return resolver.root.NodeComponents(resolver.nodeScopeContext(ctx), PaginatedQuery{Query: &query, Pagination: args.Pagination})
+	}
+	// TODO : Add postgres support
+	return nil, errors.New("Sub-resolver NodeComponents in Node does not support postgres yet")
+}
+
+// NodeComponentCount returns the number of components in the node
+func (resolver *nodeResolver) NodeComponentCount(ctx context.Context, args RawQuery) (int32, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.Nodes, "NodeComponentCount")
+	if err := readNodes(ctx); err != nil {
+		return 0, err
+	}
+
+	if !features.PostgresDatastore.Enabled() {
+		query := search.AddRawQueriesAsConjunction(args.String(), resolver.getNodeRawQuery())
+		return resolver.root.NodeComponentCount(resolver.nodeScopeContext(ctx), RawQuery{Query: &query})
+	}
+	// TODO : Add postgres support
+	return 0, errors.New("Sub-resolver NodeComponentCount in Node does not support postgres yet")
+}
+
 // TopVuln returns the first vulnerability with the top CVSS score.
 func (resolver *nodeResolver) TopVuln(ctx context.Context, args RawQuery) (VulnerabilityResolver, error) {
-	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.Nodes, "TopVulnerability")
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.Nodes, "TopVuln")
 	if err := readNodes(ctx); err != nil {
 		return nil, err
 	}
 
-	return resolver.unwrappedTopVulnQuery(ctx, args)
+	vulnResolver, err := resolver.unwrappedTopVulnQuery(ctx, args)
+	if err != nil || vulnResolver == nil {
+		return nil, err
+	}
+	return vulnResolver, nil
 }
 
-// TopVulnerability returns the first node vulnerability with the top CVSS score.
-func (resolver *nodeResolver) TopVulnerability(ctx context.Context, args RawQuery) (NodeVulnerabilityResolver, error) {
+// TopNodeVulnerability returns the first node vulnerability with the top CVSS score.
+func (resolver *nodeResolver) TopNodeVulnerability(ctx context.Context, args RawQuery) (NodeVulnerabilityResolver, error) {
 	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.Nodes, "TopVulnerability")
```

Comment:
Change the function name here

Reply:
Changed the metric name to match func name